File: Udemy course codes/ZTM/Data Structures and Algorithms/Sorting/Insertion Sort.py
num = [88, 5, 8, 3, 6, 2, 1, 7, 4, 666, 55, 21]


def insertionSort(arr):
    for i in range(1, len(arr)):
        # i>0 is used because python allows negative index
        while arr[i-1] > arr[i] and i > 0:
            temp = arr[i]
            arr[i] = arr[i-1]
            arr[i-1] = temp
            i = i-1

    return arr


print(insertionSort(num))

#tc - O(n^2)
#sc - O(1)

File: Udemy course codes/ZTM/Data Structures and Algorithms/Sorting/Insertion Sort.js
const numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0];

function insertionSort(array) {
    for (let i = 1; i < array.length; i++) {
        while (array[i - 1] > array[i]) {
            let temp = array[i]
            array[i] = array[i - 1]
            array[i - 1] = temp
            i = i - 1
        }
    }
    return array
}

insertionSort(numbers);
console.log(numbers);


//tc - O(n^2)
//sc - O(1)
